util-collection/src/main/scala/sbt/internal/util/TupleMapExtension.scala

```diff
@@ -5,28 +5,76 @@
+          t.size match
+            case 1 => t(0) :: nil
+            case 2 => t(0) :: t(1) :: nil
+            case 3 => t(0) :: t(1) :: t(2) :: nil
+            case 4 => t(0) :: t(1) :: t(2) :: t(3) :: nil
+            case 5 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: nil
+            case 6 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: nil
+            case 7 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: nil
+            case 8 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: nil
+            case 9 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: nil
+            case 10 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: nil
+            case 11 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: t(10) :: nil
+            case 12 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: t(10) :: t(11) :: nil
+            case 13 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: t(10) :: t(11) :: t(12) :: nil
+            case 14 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: t(10) :: t(11) :: t(12) :: t(13) :: nil
+            case 15 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: t(10) :: t(11) :: t(12) :: t(13) :: t(14) :: nil
+            case 16 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: t(10) :: t(11) :: t(12) :: t(13) :: t(14) :: t(15) :: nil
+            case 17 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: t(10) :: t(11) :: t(12) :: t(13) :: t(14) :: t(15) :: t(16) :: nil
+            case 18 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: t(10) :: t(11) :: t(12) :: t(13) :: t(14) :: t(15) :: t(16) :: t(17) :: nil
+            case 19 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: t(10) :: t(11) :: t(12) :: t(13) :: t(14) :: t(15) :: t(16) :: t(17) :: t(18) :: nil
+            case 20 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: t(10) :: t(11) :: t(12) :: t(13) :: t(14) :: t(15) :: t(16) :: t(17) :: t(18) :: t(19) :: nil
+            case 21 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: t(10) :: t(11) :: t(12) :: t(13) :: t(14) :: t(15) :: t(16) :: t(17) :: t(18) :: t(19) :: t(20) :: nil
+            case 22 => t(0) :: t(1) :: t(2) :: t(3) :: t(4) :: t(5) :: t(6) :: t(7) :: t(8) :: t(9) :: t(10) :: t(11) :: t(12) :: t(13) :: t(14) :: t(15) :: t(16) :: t(17) :: t(18) :: t(19) :: t(20) :: t(21) :: nil
+            case _ => tuple.iterator.toList
+      // format: on
+      list.asInstanceOf[List[F1[Any]]]
 
+    def unmap(f: [a] => F1[a] => a): Tup = transform[[A] =>> A](f).asInstanceOf[Tup]
     def transform[F2[_]](f: [a] => F1[a] => F2[a]): Tuple.Map[Tup, F2] =
```

Comment:
Thanks @adpi2 !
For future context, could you maybe leave in the original "before" code in the comment with the explanation that we need to hand roll this due to performance reason please?